models/management.cattle.io.cluster.js

```diff
@@ -227,6 +228,26 @@ export default class MgmtCluster extends HybridModel {
     return this.providerLogo;
   }
 
+  // Custom badge to show for the Cluster (if the appropriate annotations are set)
+  get badge() {
+    const text = this.metadata?.annotations[CLUSTER_BADGE.TEXT];
+
+    if (!text) {
+      return undefined;
+    }
+
+    const color = this.metadata?.annotations[CLUSTER_BADGE.COLOR];
+    const useForIcon = this.metadata?.annotations[CLUSTER_BADGE.USE_AS_ICON] || '';
+
+    return {
+      text,
+      color,
+      textColor:  textColor(parseColor(color)),
```

Comment:
This throws an exception if it fails, could catch this and track log it WRT to metadata?.annotations[CLUSTER_BADGE.COLOR]